newroutes/formsg.ts

```diff
@@ -0,0 +1,65 @@
+import { DecryptedContent, FormField } from "@opengovsg/formsg-sdk/dist/types"
+import autoBind from "auto-bind"
+import express, { Request, Response } from "express"
+
+import logger from "@logger/logger"
+
+import { BadRequestError } from "@errors/BadRequestError"
+
+import { attachFormSGHandler } from "@root/newmiddleware"
+import InfraService from "@services/infra/InfraService"
+
+const { SITE_CREATE_FORM_KEY } = process.env
+
+function getProp(responses: FormField[], name: string): string | undefined {
+  const response = responses.find(({ question }) => question === name)
+
+  return response?.answer
+}
+
+interface FormsgRouterProps {
+  infraService: InfraService
+}
+
+export class FormsgRouter {
+  private readonly infraService
+
+  constructor({ infraService }: FormsgRouterProps) {
+    this.infraService = infraService
+    // We need to bind all methods because we don't invoke them from the class directly
+    autoBind(this)
+  }
+
+  async createSite(req: Request, res: Response) {
+    const { submissionId } = req.body.data
+    const { responses } = res.locals.submission as DecryptedContent
```

Comment:
yeap! actually the types (`Request/Response`) exported by `express` have quite a number of generic parameters indicating the different components of the req/res!

so all we'd need to do here is to declare the types as follows: 
```ts
async createSite(
  req: Request<unknown, unknown, { data: { submissionId: string } }, unknown>,
  res: Response<unknown, { submission: DecryptedContent }>
  ) { ... }
```

we could also go a step further by declaring our own utility types for interfacing with express to smooth this process along in the future. do noite that because we're expecting a `res.locals.submission` here, we should ideally export the middleware together with the router or something bundling them together so that devs know when/where this gets set.

Reply:
Thanks! I made the change. It would be great if we could get these types in place for all of our routes, but I think that will have to wait for another day.